pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/ddp/abstraction/MedicalRecordAbstractionFieldDaoLive.java

```diff
@@ -0,0 +1,123 @@
+
+package org.broadinstitute.dsm.db.dao.ddp.abstraction;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A production implementation of MedicalRecordAbstractionFieldDao
+ */
+public class MedicalRecordAbstractionFieldDaoLive implements MedicalRecordAbstractionFieldDao<MedicalRecordAbstractionFieldDto> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MedicalRecordAbstractionFieldDaoLive.class);
+
+    private static final String SELECT_ALL_MEDICAL_RECORD_ABSTRACTION_FIELDS = "SELECT * FROM medical_record_abstraction_field ";
+
+    private static final String FILTER_BY_INSTANCE_ID =
+            "WHERE ddp_instance_id = (select ddp_instance_id from ddp_instance where instance_name = ?) ";
+
+    private static final String SELECT_POSSIBLE_VALUES_BY_DISPLAY_NAME_AND_TYPE = "SELECT possible_values "
+            + "FROM medical_record_abstraction_field WHERE display_name = ? AND type = ?";
```

Comment:
if it is possible to have the similar field for different studies with the same `display_name` and `type` you may end up with wrong data, I think it worths to add extra check with `ddp_instance_id` for example.